src/app/components/pages/IsaacCompetition/EntryForm/CompetitionEntryForm.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useEffect, useState } from "react";
+import { Form, Row, Col, Container, FormGroup, Label, Input } from "reactstrap";
+import { InputType } from "reactstrap/es/Input";
+import { AppGroup } from "../../../../../IsaacAppTypes";
+import { isaacApi, useAppSelector, useAppDispatch, AppDispatch } from "../../../../state";
+import { selectors } from "../../../../state/selectors";
+import { SchoolInput } from "../../../elements/inputs/SchoolInput";
+import { api } from "../../../../services";
+import { showErrorToast, showSuccessToast } from "../../../../state/actions/popups";
+
+const COMPETITON_ID = "isaac_competition_25";
+interface CompetitionEntryFormProps {
+  handleTermsClick: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => void;
+}
+
+const CompetitionEntryForm = ({ handleTermsClick }: CompetitionEntryFormProps) => {
+  const [activeGroups, setActiveGroups] = useState<AppGroup[]>([]);
+  const { data: groups } = isaacApi.endpoints.getGroups.useQuery(false);
+  const [getGroupMembers] = isaacApi.endpoints.getGroupMembers.useLazyQuery();
+  const targetUser = useAppSelector(selectors.user.orNull);
+  const [selectedGroup, setSelectedGroup] = useState<AppGroup | null>(null);
+  const dispatch: AppDispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (groups) {
+      setActiveGroups(groups);
+    }
+  }, [groups]);
+
+  useEffect(() => {
+    if (selectedGroup?.id && !selectedGroup.members) {
+      getGroupMembers(selectedGroup.id);
+    }
+  }, [selectedGroup]);
+
+  const renderFormGroup = (
+    label: string,
+    type: string,
+    id: string,
+    defaultValue: string = "",
+    options: string[] = [],
+    disabled: boolean = false,
+  ) => (
+    <FormGroup>
+      <Label className="entry-form-sub-title">{label}</Label>
+      {type === "select" ? (
+        <Input
```

Comment:
this is an example of a component that could be it's own file for the sake of cleanliness.